### PR TITLE
Unique usernames on course registration

### PR DESCRIPTION
**Pull request by hermabe**
Course registration now requires unique usernames.

### CODE REVIEW

File: make_queue/templates/make_queue/course/registration_edit.html [hunk 1]
```diff
@@ -32,6 +33,16 @@ <h1>
...
                 <p> {% trans "Are you sure you entered the correct username?" %} </p>
             </div>
+            {% form_has_error form "username" "unique" as unique_error %}
+            {% if unique_error %}
+                <div class="ui error message">
+                    <i class="close icon"></i>
+                    <div class="header">
+                        {% trans "Username already exists" %}
+                    </div>
+                    <p>{% trans "Are you sure you entered the correct username?" %}</p>
```
Comment (zootos) [inline make_queue/templates/make_queue/course/registration_edit.html]: Should we not have some other message than this in the description part of the unique constraint error. Something along the lines of "A course registration for the given username already exists. Either the user has already taken the course, or the username of another user has been entered."

File: make_queue/templates/make_queue/course/registration_edit.html [hunk 2]
```diff
@@ -32,6 +33,16 @@ <h1>
                 </div>
                 <p> {% trans "Are you sure you entered the correct username?" %} </p>
             </div>
+            {% form_has_error form "username" "unique" as unique_error %}
```
Comment (zootos) [inline make_queue/templates/make_queue/course/registration_edit.html]: I think we should use named arguments for the `field` and `code` arguments when calling the template tag. That would make the code more readable. 

```{% form_has_error form field="username" code="unique" as unique_error %}```

File: web/templatetags/util_tags.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from django import template
+
+register = template.Library()
+
+
+@register.simple_tag()
+def form_has_error(form, field, code):
```
Comment (zootos) [inline web/templatetags/util_tags.py]: Rename `code` argument to `error_code` for better readability.

File: make_queue/migrations/0010_auto_20190302_2210.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 2.1.5 on 2019-03-02 21:10
+
+from django.db import migrations
+import make_queue.models.course
```
Comment (zootos) [inline make_queue/migrations/0010_auto_20190302_2210.py]: Rename file to `0010_unique_username_in_course_registration.py` to make it easier to see what migrations does (I just recently found out that you could rename the migrations, and think it makes sense to do this for all future migrations).

File: web/templatetags/util_tags.py [hunk 5]
```diff
@@ -0,0 +1,8 @@
+from django import template
+
+register = template.Library()
```
Comment (zootos) [inline web/templatetags/util_tags.py]: Rename file to `form_utility.py`?

File: make_queue/migrations/0010_auto_20190302_2210.py (hunk 4 above)
Comment (hermabe) [reply make_queue/migrations/0010_auto_20190302_2210.py]: Yes, that is useful.

File: make_queue/templates/make_queue/course/registration_edit.html (hunk 1 above)
Comment (hermabe) [reply make_queue/templates/make_queue/course/registration_edit.html]: Yes, that gives the user more information.

File: locale/nb/LC_MESSAGES/django.po [hunk 6]
```diff
@@ -1452,3 +1459,6 @@ msgstr "Velg verdi"
 #: web/widgets.py:34
 msgid "Search places"
 msgstr "Søk etter steder"
+
+#~ msgid "Username already exists"
```
Comment (zootos) [inline locale/nb/LC_MESSAGES/django.po]: Can we delete this unused translation

Comment (hermabe) [reply locale/nb/LC_MESSAGES/django.po]: Yes

**Pull request merged**